File: patterns-java/src/test/java/me/freedom4live/patterns/java/factory/impl/CatFactoryImplTest.java
package me.freedom4live.patterns.java.factory.impl;

import me.freedom4live.patterns.java.factory.Cat;
import me.freedom4live.patterns.java.factory.CatFactory;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class CatFactoryImplTest {

    private CatFactory catFactory;


    @BeforeEach
    void setUp() {
        catFactory = new CatFactoryImpl();
    }

    @Test
    void createRandomCat_normal_createdAnInstance() {
        //act
        Cat cat = catFactory.createRandomCat();

        //assert
        Assertions.assertNotNull(cat);
    }

    @Test
    void createNamedCat_giveAName_createdAnInstance() {
        //arrange
        String name = "TEST_NAME";

        //act
        Cat cat = catFactory.createNamedCat(name);

        //assert
        Assertions.assertNotNull(cat);
    }

    @Test
    void createRandomCat_normal_meowRandomly() {
        //act
        Cat cat = catFactory.createRandomCat();

        //assert
        Assertions.assertNotNull(cat);
        Assertions.assertNotNull(cat.meow());
    }

    @Test
    void createNamedCat_giveAName_meowGivenName() {
        //arrange
        String name = "TEST_NAME";

        //act
        Cat cat = catFactory.createNamedCat(name);

        //assert
        Assertions.assertNotNull(cat);
        Assertions.assertEquals(name, cat.meow());
    }

}

File: patterns-kotlin/src/test/kotlin/me/freedom4live/patterns/kotlin/factory/impl/CatFactoryImplTest.kt
package me.freedom4live.patterns.kotlin.factory.impl

import me.freedom4live.patterns.kotlin.factory.CatFactory
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class CatFactoryImplTest {

    private lateinit var catFactory: CatFactory


    @BeforeEach
    fun setUp() {
        catFactory = CatFactoryImpl()
    }

    @Test
    fun randomCat_normal_instanceIsNotNull() {
        //act
        val cat = catFactory.randomCat()

        //assert
        Assertions.assertNotNull(cat)
    }

    @Test
    fun namedCat_name_isNotNull() {
        //arrange
        val name = "TEST_NAME"

        //act
        val cat = catFactory.namedCat(name)

        //assert
        Assertions.assertNotNull(cat)
    }

    @Test
    fun randomCat_normal_randomMeow() {
        //act
        val cat = catFactory.randomCat()

        //assert
        Assertions.assertNotNull(cat)
        Assertions.assertNotNull(cat.meow())
    }

    @Test
    fun namedCat_name_meowItsName() {
        //arrange
        val name = "TEST_NAME"

        //act
        val cat = catFactory.namedCat(name)

        //assert
        Assertions.assertNotNull(cat)
        Assertions.assertEquals(name, cat.meow())
    }

}
